src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/TextReplacement.cs

```diff
@@ -78,8 +80,10 @@ public TextReplacement(SourceText originalText, SourceText newText, string fileP
         public override int GetHashCode()
         {
             var hashcode = default(HashCode);
+            hashcode.Add(StartingLine);
+            hashcode.Add(FilePath, StringComparer.OrdinalIgnoreCase);
             hashcode.Add(NewText.ToString(), StringComparer.Ordinal);
```

Comment:
calling `.ToString` seems like it may be a potentially heavy operation. Does NewText not implement `GetHashCode()`?

Reply:
Hm.. that's a good point. I don't think `SourceText` (which NewText is) has a GetHashCode, though, since it doesn't implement `IEquatable`.  Now that  look at the code, though, I'll just make the property a string type. We're not using any `SourceText` APIs (the type is just an artifact of where the data comes from in UA's workflow). That way, at least, we won't have multiple ToString calls.